File: BackendSilky/src/main/java/com/proweb/Silky/model/Purchase.java
package com.proweb.Silky.model;

import java.time.LocalDateTime;
import java.util.List;

import javax.persistence.Basic;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Entity
@Table(name = "purchases")
public class Purchase {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	@Basic
	private Float total;
	@Basic
	private LocalDateTime date;
	@ManyToOne
	private User user;
	@Column(nullable= false)
	@OneToMany(fetch = FetchType.EAGER)
	private List<PurchaseItem> purchaseItems;

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public Float getTotal() {
		return total;
	}

	public void setTotal(Float total) {
		this.total = total;
	}

	public LocalDateTime getDate() {
		return date;
	}

	public void setDate(LocalDateTime date) {
		this.date = date;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public List<PurchaseItem> getPurchaseItems() {
		return purchaseItems;
	}

	public void setPurchaseItems(List<PurchaseItem> purchaseItems) {
		this.purchaseItems = purchaseItems;
	}

	@Override
	public String toString() {
		return "Purchase [id=" + id + ", total=" + total + ", date=" + date + ", user=" + user + ", purchaseItems="
				+ purchaseItems + "]";
	}

}

File: FrontendSilky/src/app/shared/models/purchase.ts
import { PurchaseItem } from './purchaseitem';
import { User } from './user';

export class Purchase {
  public static numPur: number = 0 ;
  public purchaseItems: PurchaseItem[];
  public total: number;
  public date: Date;
  public id: number;
  public user: User;


  public get _id(): number {
    return this.id;
  }
  public set _id(value: number) {
    this.id = value;
  }


  constructor(){
    this.id = Purchase.numPur;
    Purchase.numPur++;
    this.date = new Date(Date.now());
    this.purchaseItems= [];

  }
  public get _purchaseItems(): PurchaseItem[] {
    return this.purchaseItems;
  }
  public set _purchaseItems(value: PurchaseItem[]) {
    this.purchaseItems = value;
  }
  public get _total(): number {
    return this.total;
  }
  public set _total(value: number) {
    this.total = value;
  }
  public get _date(): Date {
    return this.date;
  }
  public set _date(value: Date) {
    this.date = value;
  }
  public get _user(): User {
    return this.user;
  }
  public set _user(value: User) {
    this.user = value;
  }


}
